app/lib/features/home/widgets/custom_selected_icon.dart

```diff
@@ -1,21 +1,32 @@
+import 'package:acter/features/home/data/keys.dart';
+import 'package:acter/features/news/providers/news_providers.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
 
-class CustomSelectedIcon extends StatelessWidget {
+class CustomSelectedIcon extends ConsumerWidget {
   final Widget icon;
+
   const CustomSelectedIcon({super.key, required this.icon});
 
   @override
-  Widget build(BuildContext context) {
-    return Container(
-      height: 30,
-      width: 54,
-      margin: const EdgeInsets.symmetric(horizontal: 8),
-      padding: const EdgeInsets.only(bottom: 2),
-      decoration: BoxDecoration(
-        color: Theme.of(context).colorScheme.primary,
-        borderRadius: BorderRadius.circular(14),
+  Widget build(BuildContext context, WidgetRef ref) {
+    return InkWell(
+      onDoubleTap: () {
+        if (key == MainNavKeys.updates) {
+          ref.read(boostGoToPageProvider.notifier).state = 0;
```

Comment:
This code line supposed to trigger `ref.listenManual(boostGoToPageProvider,..` in news_full_view.dart as shown below.
@gnunicorn

Reply:
but when is it `!= 0`? if you set the state to what it already is, riverpod will not propagate this as an update.

Reply:
Ahh, I see. Thank you so much for the help.